.github/workflows/on-pr.yml

```diff
@@ -16,3 +16,49 @@ jobs:
     uses: ./.github/workflows/build-device.yml
     with:
       timeout: 10
+
+  pre-commit:
+    name: Run Pre-commit Hooks and Propose Fixes
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history so 'origin/main' is available
+          ref: ${{ github.head_ref }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+
+      - name: Run Pre-commit and Fix Issues
```

Comment:
I see this in logs:

gersemi..............................................(no files to check)Skipped
clang-format.........................................(no files to check)Skipped

Can you try to submit a wrongly formatted cmake and/or cpp file, and see if the PR will be blocked?

Reply:
Already did this earlier in this pr history

Reply:
It will work, I guarantee

Reply:
Just added change to C++

Reply:
Ah, looks like it checks only touched cpp files.
Thanks for checking!